### PR TITLE
Fixes for Multi-sample, implementation of MuData and Docs for hashing

### PR DESCRIPTION
**Pull request by mari-ga**
* Multiple fixes for multi-sample hashing
* implementation of MuData for summary hashing demultiplexing
* Added docs for new tools hashing (GMM-Demux, BFF and Demuxmix)
* Other fixes (correction of parameter for BFF)

### CODE REVIEW

**Review by wxicu — CHANGES_REQUESTED** [review]

File: modules/hash_demulti/preprocess.nf [hunk 1]
```diff
@@ -12,10 +15,9 @@ process preprocess{
         val number_features
         val assay
         val margin
-        val normalisation_method
-        each rna_available
-        each raw_data_object
+        val normalisation_methodx
```
[the remark this answers is not in the document]
Comment (mari-ga) [reply modules/hash_demulti/preprocess.nf:18]: Yes, deleted x on next update

File: modules/hash_demulti/preprocess.nf [hunk 2]
```diff
@@ -43,11 +43,11 @@ workflow preprocessing_hashing{
         assay = params.assay
         margin = params.margin
         norm_method = params.norm_method
-        rna_available = split_input(params.rna_available)
-        demuxmix_preprocess_mode = split_input(params.demuxmix_preprocess_mode)
         out_file = params.preprocessOut
-        preprocess(input_list, hto_raw_or_filtered, rna_raw_or_filtered, ndelim, sel_method, n_features, assay, margin, norm_method,rna_available,demuxmix_preprocess_mode,out_file)
+        gene_col = params.gene_col
+
+        preprocess(input_list, hto_raw_or_filtered, rna_raw_or_filtered,sel_method, ndelim, n_features, assay, margin, norm_method, out_file, gene_col)
```
[the remark this answers is not in the document]
Comment (mari-ga) [reply modules/hash_demulti/preprocess.nf:49]: simply checking the order or arguments, back in original order

**Comment by wxicu** [comment]
Hey @mari-ga sorry I didnt see this earlier, is this ready for review now?

**Comment by Zethson** [comment]
@mari-ga what's the status of this PR?

File: bin/summary_hash.py [hunk 3]
```diff
@@ -46,12 +47,19 @@ def demuxem_summary(demuxem_res, raw_adata, raw_mudata):
             adata.obs.donor = adata.obs.donor.astype(str)
             adata.write("hash_summary/adata/adata_with_"+os.path.basename(x)+".h5ad")
         
-        # TODO: add mudata
+        if raw_mudata is not None:
```
Comment (wxicu) [inline bin/summary_hash.py:50]: also add HTO please

File: bin/summary_hash.py [hunk 4]
```diff
@@ -46,12 +47,19 @@ def demuxem_summary(demuxem_res, raw_adata, raw_mudata):
...
+            mudata = raw_mudata.copy()
+            mudata['rna'].obs = mudata['rna'].obs.merge(demuxem_assign, left_index=True, right_index=True, how='left')
+            mudata['rna'].obs.rename(columns={mudata['rna'].obs.columns[0]: 'donor'}, inplace=True)
+            mudata['rna'].obs.donor = mudata['rna'].obs.donor.fillna("negative")
+            mudata['rna'].obs.donor = mudata['rna'].obs.donor.astype(str)
+            mudata.update()
+            mudata.write("hash_summary/mudata/mudata_with_mudata_"+ os.path.basename(x)+".h5mu")
```
Comment (wxicu) [inline bin/summary_hash.py:57]: `mudata_with` instead of `mudata_with_mudata_`

File: nextflow.config [hunk 5]
```diff
@@ -434,9 +430,11 @@ process {
         conda = "$projectDir/conda/dropletutils.yml"
     }
     withName:hash_solo {
+
```
Comment (wxicu) [inline nextflow.config:433]: please remove empty lines

File: nextflow.config [hunk 6]
```diff
@@ -405,21 +403,19 @@ profiles{
...
-                memory = 64.GB
-            }
-            withLabel: small_mem {
-                cpus = 16
-                memory = 32.GB
-            }
-        }
-    }
+         process {
+             executor = 'local'
```
Comment (wxicu) [inline nextflow.config:407]: executor should be slurm

File: modules/hash_demultiplexing.nf [hunk 7]
```diff
@@ -170,30 +167,32 @@ workflow hash_demultiplexing{
...
     }
+
     if (params.bff == "True"){
         Channel.fromPath(params.multi_input) \
                 | splitCsv(header:true) \
                 | map { row-> tuple(row.sampleId, params.hto_matrix_bff == "raw" ? row.hto_matrix_raw : row.hto_matrix_filtered )}
                 | bff_hashing
-        bff_out= bff_hashing.out
+        bff_out = bff_hashing.out
+        print("BFF path to output")
```
Comment (wxicu) [inline modules/hash_demultiplexing.nf:195]: why print this out?

**Review by wxicu — CHANGES_REQUESTED** [review]
Thanks @mari-ga for the contribution! The code looks good, just need some minor changes

**Pull request merged**